File: 0024.swap_nodes_in_pairs/solution.py
class ListNode:
    def __init__(self, x)
        self.x = x
        self.next = None


class Solution:
    def swap_pairs(self, head: ListNode) -> ListNode:
        """ 每次向后移动两位, 交换前后节点值即可 """"
        p = head
        while p and p.next:
            p.val, p.next.val = p.next.val, p.val
            p = p.next.next
        return head

File: 0024.swap_nodes_in_pairs/solution.go
package leetcode0024

import "github.com/Ackerr/Algorithms/utils"

func swapPairs(head *utils.ListNode) *utils.ListNode {
	p := head
	for p != nil && p.Next != nil {
		p.Val, p.Next.Val = p.Next.Val, p.Val
		p = p.Next.Next
	}
	return head
}
